### PR TITLE
Initial work on misc2020.

### LINKED ISSUE #11: Copy over Misc2019 folder from last year's code.
- [x] ButtonToggleRunner
- [x] EnhancedJoystick
- [ ] FunctionRunnerSendable
- [x] Gamepad

### PR DESCRIPTION
**Pull request by TaeSeanDo**
SendableBase, PIDSource, and PIDSourceType are depreciated. Closes #11.

### CODE REVIEW

**Comment by docprofsky** [comment]
I do not want us to use depreciated code, so let us read the docs on the new way to do this.
This page details many of the changes to WPI Lib.
https://docs.wpilib.org/en/latest/docs/software/wpilib-overview/new-for-2020.html

**Comment by TaeSeanDo** [comment]
Right I figured we wouldn't want to use depreciated code. I will look into it. Thanks for the link.

File: src/main/java/frc/misc2020/FunctionRunnerSendable.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+
+/**
+ * Use this class to have a button on the SmartDashboard that will run a
+ * function.
+ * 
+ * @author schuyler
+ *
+ */
+public class FunctionRunnerSendable extends SendableBase implements Sendable {
```
Comment (TaeSeanDo) [inline src/main/java/frc/misc2020/FunctionRunnerSendable.java]: This still needs to be updated to use SendableRegistry instead of SendableBase.

Comment (docprofsky) [reply src/main/java/frc/misc2020/FunctionRunnerSendable.java]: Can a student be assigned to work on this?

File: src/main/java/frc/misc2020/EnhancedJoystick.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        if (value > deadbandSize) {
+            return (value - deadbandSize) / (1 - deadbandSize);
+        }
+        if (value < -deadbandSize) {
+            return (value + deadbandSize) / (1 - deadbandSize);
+        }
+        return 0;
+
+    }
+
```
Comment (docprofsky) [inline src/main/java/frc/misc2020/EnhancedJoystick.java]: Fix this whitespace.

File: src/main/java/frc/misc2020/FunctionRunnerSendable.java (hunk 1 above)
Comment (docprofsky) [reply src/main/java/frc/misc2020/FunctionRunnerSendable.java]: This needs to be fixed before I will approve.

File: src/main/java/frc/misc2020/Gamepad.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package frc.misc2020;
+
+public class Gamepad extends EnhancedJoystick {
```
Comment (docprofsky) [inline src/main/java/frc/misc2020/Gamepad.java]: Do we want to add javadoc?

Comment (TaeSeanDo) [reply src/main/java/frc/misc2020/Gamepad.java]: Sure. (later)

**Pull request merged**